File: frontend/src/types/Announcement.ts
export interface Announcement {
  id: string;
  userId: string;
  organizationId?: string;
  userName: string;
  userProfilePicUrl?: string;
  userRole: string;
  title: string;
  content: string;
  imageUrl?: string;
  isPinned: boolean;
  category: AnnouncementCategory;
  createdAt: string;
  updatedAt: string;
}

export type AnnouncementCategory = 
  | 'GENERAL'
  | 'WORSHIP'
  | 'EVENTS'
  | 'MINISTRY'
  | 'YOUTH'
  | 'MISSIONS'
  | 'PRAYER'
  | 'COMMUNITY'
  | 'URGENT'
  | 'CELEBRATION';

export interface AnnouncementCreateRequest {
  title: string;
  content: string;
  imageUrl?: string;
  category: AnnouncementCategory;
  isPinned?: boolean;
  isSystemWide?: boolean; // For PLATFORM_ADMIN: create system-wide announcement visible to all organizations
}

export interface AnnouncementUpdateRequest {
  title: string;
  content: string;
  imageUrl?: string;
  category: AnnouncementCategory;
  isPinned?: boolean;
  isSystemWide?: boolean; // For PLATFORM_ADMIN: update to system-wide announcement
}

export interface AnnouncementResponse {
  content: Announcement[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
  first: boolean;
  last: boolean;
  empty: boolean;
}

export interface AnnouncementStats {
  totalAnnouncements: number;
  pinnedAnnouncements: number;
}

File: backend/src/main/java/com/churchapp/entity/Announcement.java
package com.churchapp.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "announcements", indexes = {
    @Index(name = "idx_announcement_created_at", columnList = "created_at"),
    @Index(name = "idx_announcement_category", columnList = "category"),
    @Index(name = "idx_announcement_pinned", columnList = "is_pinned"),
    @Index(name = "idx_announcement_user_id", columnList = "user_id"),
    @Index(name = "idx_announcements_organization_id", columnList = "organization_id")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Announcement {
    
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id", updatable = false, nullable = false)
    private UUID id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @NotBlank(message = "Title is required")
    @Size(min = 5, max = 200, message = "Title must be between 5 and 200 characters")
    @Column(name = "title", nullable = false, length = 200)
    private String title;
    
    @NotBlank(message = "Content is required")
    @Size(min = 10, max = 5000, message = "Content must be between 10 and 5000 characters")
    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;
    
    @Column(name = "image_url", length = 500)
    private String imageUrl;
    
    @Column(name = "is_pinned", nullable = false)
    private Boolean isPinned = false;
    
    @Enumerated(EnumType.STRING)
    @Column(name = "category", nullable = false)
    private AnnouncementCategory category = AnnouncementCategory.GENERAL;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    @Column(name = "deleted_at")
    private LocalDateTime deletedAt;

    // Multi-tenant organization field
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id")
    private Organization organization;

    public enum AnnouncementCategory {
        GENERAL,
        WORSHIP,
        EVENTS,
        MINISTRY,
        YOUTH,
        MISSIONS,
        PRAYER,
        COMMUNITY,
        URGENT,
        CELEBRATION
    }
    
    // Soft delete helper methods
    public boolean isDeleted() {
        return deletedAt != null;
    }
    
    public void markAsDeleted() {
        this.deletedAt = LocalDateTime.now();
    }
    
    public void restore() {
        this.deletedAt = null;
    }
}